### PR TITLE
Support for LspSaga

### PR DESCRIPTION
**Pull request by ExtinctAxolotl**
Hey there!
Thank you all for developing this theme!

I recently started using [lspsaga](https://github.com/tami5/lspsaga.nvim) and am very enjoying it functionality wise.
Problem is, lspsaga defines its own highlight groups, so it looks a bit off compared to the rest of the colorscheme.

image

> Here again the borders and I guess the weird black line:
image

> And so on...
image

## Where is this going
I will try in the next three weeks or so to add full support for it.

Also this is the first time I am doing a PR on anything really so any help is greatly appreciated 🙂 (and patience too)

### CODE REVIEW

**Comment by mvllow** [comment]
How exciting! Let us know if you have troubles with anything – one tip to start would be to test locally by adding your local folder to your plugin manager. Eg. using packer

```diff
- use('rose-pine/neovim')
+ use('~/your-folder')
```

Then a `:PackerSync` and you're all set 😌 Now changes to your folder will reflect in your editor. Hope that helps!

**Comment by ExtinctAxolotl** [comment]
image
> This is the new diagnostic preview...
image
> ... and this the code actions prompt

**Comment by ExtinctAxolotl** [comment]
Most colours are now added, so I'm going to test this now.

**Comment by ExtinctAxolotl** [comment]
just the definition preview that has something wrong with it...

**Comment by ExtinctAxolotl** [comment]
image
🤔

**Comment by ExtinctAxolotl** [comment]
Ready for review!

File: lspsaga.norg [hunk 1]
```diff
@@ -0,0 +1,93 @@
+@document.meta
```
Comment (mvllow) [inline lspsaga.norg:1]: Can we remove this file?

File: lua/rose-pine/theme.lua [hunk 2]
```diff
@@ -410,6 +410,49 @@ local theme = {
 	VimwikiLink = { fg = p.rose },
 	VimwikiList = { fg = p.iris },
 	VimwikiNoExistsLink = { fg = p.love },
+
+	-- LspSaga
+	-- https://github.com/tami5/lspsaga.nvim (fork of https://github.com/glepnir/lspsaga.nvim)
+	DiagnosticInformation = { link = 'DiagnosticInfo' },
+	DiagnosticWarning = { link = 'DiagnosticWarn' },
```
Comment (mvllow) [inline lua/rose-pine/theme.lua:417]: Let's move these near there respective spots, eg. put `DiagnosticWarning` above our `DiagnosticWarn`

File: lua/rose-pine/theme.lua [hunk 3]
```diff
@@ -410,6 +410,49 @@ local theme = {
...
+    LspSagaAutoPreview = { fg = p.subtle },
+    LspSagaDefPreviewBorder = { fg = p.subtle },
+	DefinitionIcon = { fg = p.rose },
+    DefinitionCount = { fg = p.rose },
+    ReferencesCount = { fg = p.rose },
+    ReferencesIcon = { fg = p.rose },
+    DefinitionPreviewTitle = {
+        fg = p.rose,
+        style = "bold",
+    },
```
Comment (mvllow) [inline lua/rose-pine/theme.lua:443]: Are these groups provided by LspSaga? If not, they should be moved to a more generic location.

File: lua/rose-pine/theme.lua [hunk 4]
```diff
@@ -410,6 +410,49 @@ local theme = {
...
+	LspSagaRenamePromptPrefix = { fg = p.love },
+	LspSagaRenameBorder = { fg = p.pine },
+    LspFloatWinNormal = { bg = p.base },
+    LspFloatWinBorder = { fg = p.subtle },
+    LspSagaDocTruncateLine = { link = "LspSagaHoverBorder" },
+    LspSagaHoverBorder = { fg = p.subtle},
+    LspSagaSignatureHelpBorder = { fg = p.pine },
+    LspSagaShTruncateLine = { link = "LspSagaSignatureHelpBorder" },
+    TargetWord = { fg = p.iris },
```
Comment (mvllow) [inline lua/rose-pine/theme.lua:452]: Are these groups provided by LspSaga? If not, they should be moved to a more generic location.

**Review by mvllow — CHANGES_REQUESTED** [review]
After making changes, please format with stylua 😌 Thank you!

File: lspsaga.norg (hunk 1 above)
Comment (ExtinctAxolotl) [reply lspsaga.norg:1]: Yep! I used this to sort of get an overview of all things I need to do, but then accidentally committed it.

File: lua/rose-pine/theme.lua (hunk 2 above)
Comment (ExtinctAxolotl) [reply lua/rose-pine/theme.lua:417]: Alrighty

File: lua/rose-pine/theme.lua (hunk 3 above)
Comment (ExtinctAxolotl) [reply lua/rose-pine/theme.lua:442]: Yes, these are provided by LspSaga and not used anywhere else.

File: lua/rose-pine/theme.lua (hunk 4 above)
Comment (ExtinctAxolotl) [reply lua/rose-pine/theme.lua:451]: Yes, these are provided by LspSaga and not used anywhere else.

**Comment by ExtinctAxolotl** [comment]
Hope this is good :grimacing:...

**Comment by ExtinctAxolotl** [comment]
I don't have a good feeling about this... sorry!

**Comment by mvllow** [comment]
No worries! I'll take a look a little later today and we'll get it all taken care of 😌

**Comment by ExtinctAxolotl** [comment]
Thank you! :heart:

File: lua/rose-pine/theme.lua [hunk 5]
```diff
@@ -176,9 +176,7 @@ local theme = {
 
 	DiagnosticHint = { fg = config.colors.hint },
 	DiagnosticInfo = { fg = config.colors.info },
-	DiagnosticInformation = { link = 'DiagnosticInfo' },
 	DiagnosticWarn = { fg = config.colors.warn },
-	DiagnosticWarning = { link = 'DiagnosticWarn' },
```
Comment (mvllow) [inline lua/rose-pine/theme.lua]: Did you mean to remove these?

**Review by mvllow — COMMENTED** [review]
Please use [stylua](https://github.com/JohnnyMorganz/StyLua#installation) to format

File: lua/rose-pine/theme.lua (hunk 5 above)
Comment (ExtinctAxolotl) [reply lua/rose-pine/theme.lua]: Nope, readded them now...

**Comment by ExtinctAxolotl** [comment]
That is everything done?

**Review by mvllow — APPROVED** [review]

**Comment by mvllow** [comment]
Not sure why stylua didn't pick up our formatting rules but not a problem, everything looks good now! Thanks for all the hard work 😌

**Comment by ExtinctAxolotl** [comment]
Thank you for your kindness and patience @mvllow!

**Pull request merged**